### PR TITLE
Raincatch 756 return error for no document found

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
Ensuring that when reading a single document, if it is not found then it is an error case.

### CODE REVIEW

File: lib/mongoose-store.js [hunk 1]
```diff
@@ -45,13 +57,20 @@ Store.prototype.findById = function(id) {
 };
 
 Store.prototype.read = function(_id) {
-  return this.model.findOne({id: _id}).exec().then(convertToJSON);
+  return this.model.findOne({id: _id}).exec().then(function(foundDocument) {
```
Comment (wtrocki) [inline lib/mongoose-store.js:78]: We missing catch here. I think that nothing will be returned if other error happened (lost database connection, backend overloaded etc.)? Best to supply catch handler as well.

**Review by wtrocki — APPROVED** [review]
Works great! Verified on local machine

File: lib/mongoose-store.js [hunk 2]
```diff
@@ -45,13 +57,20 @@ Store.prototype.findById = function(id) {
...
+
+    return foundDocument;
+  }).then(convertToJSON);
 };
 
 Store.prototype.update = function(object) {
   return this.model.findOne({id: object.id}).exec().then(function(foundDocument) {
     if (!foundDocument) {
-      return Promise.reject(new Error("No document with id " + object.id  + " found"));
+      return createNoDocumentError(object.id);
```
Comment (wtrocki) [inline lib/mongoose-store.js:93]: [Question] Should update attempt to save in this case?

Comment (nialldonnellyfh) [reply lib/mongoose-store.js:93]: I don't think so. If someone is trying to update something that isn't there it should be an error.

File: lib/mongoose-store.js (hunk 1 above)
Comment (nialldonnellyfh) [reply lib/mongoose-store.js:78]: Yes, good point.

**Comment by ghost** [comment]
LGTM

**Pull request merged**